logicrunner/handle_additional_call.go

```diff
@@ -31,54 +31,70 @@ import (
 	"github.com/insolar/insolar/instrumentation/instracer"
 )
 
-type HandleAdditionalCallFromPreviousExecutor struct {
-	dep *Dependencies
+type AdditionalCallFromPreviousExecutor struct {
+	stateStorage StateStorage
 
-	Message payload.Meta
-	Parcel  insolar.Parcel
+	message *message.AdditionalCallFromPreviousExecutor
 }
 
-// This is basically a simplified version of HandleCall.handleActual().
-// Please note that currently we lack any fraud detection here.
-// Ideally we should check that the previous executor was really an executor
-// during previous pulse, that the request was really registered, etc.
-// Also we don't handle case when pulse changes during execution of this handle.
-// In this scenario user is in a bad luck. The request will be lost and user will have
-// to re-send it after some timeout.
-func (h *HandleAdditionalCallFromPreviousExecutor) handleActual(
-	ctx context.Context,
-	msg *message.AdditionalCallFromPreviousExecutor,
-	_ flow.Flow,
-) {
-	broker := h.dep.StateStorage.UpsertExecutionState(msg.ObjectReference)
-
-	if msg.Pending == insolar.NotPending {
+func (p *AdditionalCallFromPreviousExecutor) Proceed(ctx context.Context) error {
+	broker := p.stateStorage.UpsertExecutionState(p.message.ObjectReference)
+
+	if p.message.Pending == insolar.NotPending {
 		broker.SetNotPending(ctx)
 	}
 
-	tr := NewTranscript(freshContextFromContext(ctx), msg.RequestRef, msg.Request)
+	tr := NewTranscript(freshContextFromContext(ctx), p.message.RequestRef, p.message.Request)
 	broker.AddAdditionalRequestFromPrevExecutor(ctx, tr)
+	return nil
+}
+
+type HandleAdditionalCallFromPreviousExecutor struct {
+	dep *Dependencies
+
+	Message payload.Meta
+	Parcel  insolar.Parcel
 }
 
+// Please note that currently we lack any fraud detection here.
+// Ideally we should check that the previous executor was really an executor during previous pulse,
+// that the request was really registered, etc. Also we don't handle case when pulse changes during
+// execution of this handle. In this scenario user is in a bad luck. The request will be lost and
+// user will have to re-send it after some timeout.
 func (h *HandleAdditionalCallFromPreviousExecutor) Present(ctx context.Context, f flow.Flow) error {
 	ctx = loggerWithTargetID(ctx, h.Parcel)
-	inslogger.FromContext(ctx).Debug("HandleAdditionalCallFromPreviousExecutor.Present starts ...")
+
+	logger := inslogger.FromContext(ctx).WithField("handler", "HandleAdditionalCallFromPreviousExecutor")
+	logger.Debug("Handler.Present starts")
 
 	msg := h.Parcel.Message().(*message.AdditionalCallFromPreviousExecutor)
-	ctx = contextFromServiceData(msg.ServiceData)
+	ctx = contextWithServiceData(ctx, msg.ServiceData)
```

Comment:
this looks wrong. it's better to move old two lines into beginning of the function